router/pkg/mcpserver/server.go

```diff
@@ -0,0 +1,676 @@
+type operationHandler struct {
+	operation      schemaloader.Operation
+	compiledSchema *jsonschema.Schema
+}
+
+// OperationInfo contains metadata about a GraphQL operation
+type OperationInfo struct {
+	Name        string          `json:"name"`
+	Description string          `json:"description"`
+	Schema      json.RawMessage `json:"schema,omitempty"`
+	Query       string          `json:"query"`
+}
+
+// OperationsResponse is the response structure for the listGraphQLOperations tool
+type OperationsResponse struct {
+	Operations  []OperationInfo `json:"operations"`
+	Usage       string          `json:"usage"`
+	LLMGuidance LLMGuidance     `json:"llmGuidance"`
+	Endpoint    string          `json:"endpoint"`
+}
+
+// ListOperationsResponse is the response structure for the list_graphql_operations tool.
+type ListOperationsResponse struct {
+	Operations []SimpleOperationInfo `json:"operations"`
+	LLMNote    string                `json:"llmNote"`
+}
+
+// GraphQLOperationInfoResponse is the response structure for the graphql_operation_info tool.
+type GraphQLOperationInfoResponse struct {
+	Name           string          `json:"name"`
+	Description    string          `json:"description"`
+	OperationType  string          `json:"operationType"`
+	HasSideEffects bool            `json:"hasSideEffects"`
+	Schema         json.RawMessage `json:"schema,omitempty"`
+	Query          string          `json:"query"`
+	LLMGuidance    LLMGuidance     `json:"llmGuidance"`
+	Endpoint       string          `json:"endpoint"`
+}
+
+// GraphQLOperationInfoInput defines the input structure for the graphql_operation_info tool.
+type GraphQLOperationInfoInput struct {
+	OperationName string `json:"operationName"`
+}
+
+// LLMGuidance provides guidance for LLMs on how to use the GraphQL operations
+type LLMGuidance struct {
+	HTTPUsage      string   `json:"httpUsage"`
+	GraphQLRequest string   `json:"graphqlRequest"`
+	ExecutionTips  []string `json:"executionTips"`
+}
+
+// NewGraphQLSchemaServer creates a new GraphQL schema server
+func NewGraphQLSchemaServer(routerGraphQLEndpoint string, schema *ast.Document, opts ...func(*Options)) (*GraphQLSchemaServer, error) {
+
+	if routerGraphQLEndpoint == "" {
+		return nil, fmt.Errorf("routerGraphQLEndpoint cannot be empty")
+	}
+
+	if schema == nil {
+		return nil, fmt.Errorf("schema cannot be nil")
+	}
+
+	if !strings.Contains(routerGraphQLEndpoint, "://") {
+		routerGraphQLEndpoint = "http://" + routerGraphQLEndpoint
+	}
+
+	// Default options
+	options := &Options{
+		GraphName:     "graph",
+		OperationsDir: "operations",
+		Server: ServerOptions{
+			Port: 5025,
+		},
+		Enabled:        false,
+		Logger:         zap.NewNop(),
+		RequestTimeout: 30 * time.Second,
+		ExposeSchema:   true,
+	}
+
+	// Apply all option functions
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	// Create the MCP server
+	mcpServer := server.NewMCPServer(
+		"wundergraph-cosmo-"+options.GraphName,
```

Comment:
Should we allow the user to define the name and version of their MCP server?

Side note, should we think in our future design to allow supporting multiple versions?

Reply:
I think allowing a suffix is a good starting point. By default it fallbacks to `wundergraph-cosmo-graph`.

> Side note, should we think in our future design to allow supporting multiple versions?

Not for now. The version should be sticked to the version of the router.